modules/drivers/mongo/src/metabase/driver/mongo.clj

```diff
@@ -200,65 +192,45 @@
+    (let [collection (mongo.util/collection db (:name table))]
+      (->> (mongo.util/list-indexes collection :keywordize false)
+           (map (fn [index]
                 ;; for text indexes, column names are specified in the weights
-                (if (contains? index "textIndexVersion")
-                  (get index "weights")
-                  (get index "key"))))
-         (map (comp name first keys))
-         ;; mongo support multi key index, aka nested fields index, so we need to split the keys
-         ;; and represent it as a list of field names
-         (map #(if (str/includes? % ".")
-                 {:type  :nested-column-index
-                  :value (str/split % #"\.")}
-                 {:type  :normal-column-index
-                  :value %}))
-         set)))
-
-(defn- from-db-object
-  "This is mostly a copy of the monger library's own function of the same name with the
-  only difference that it uses an ordered map to represent the document. This ensures that
-  the order of the top level fields of the table is preserved. For anything that's not a
-  DBObject, it falls back to the original function."
-  [input]
-  (if (instance? DBObject input)
-    (let [^DBObject dbobj input]
-      (reduce (fn [m ^String k]
-                (assoc m (keyword k) (m.conversion/from-db-object (.get dbobj k) true)))
-              (ordered-map/ordered-map)
-              (.keySet dbobj)))
-    (m.conversion/from-db-object input true)))
-
-(defn- sample-documents [^com.mongodb.DB conn table sort-direction]
-  (let [collection (.getCollection conn (:name table))]
-    (with-open [cursor (doto (.find collection
-                                    (m.conversion/to-db-object {})
-                                    (m.conversion/as-field-selector []))
-                         (.limit metadata-queries/nested-field-sample-limit)
-                         (.skip 0)
-                         (.sort (m.conversion/to-db-object {:_id sort-direction}))
-                         (.batchSize 256))]
-      (map from-db-object cursor))))
+                  (if (contains? index "textIndexVersion")
+                    (get index "weights")
+                    (get index "key"))))
+           (map (comp name first keys))
+           ;; mongo support multi key index, aka nested fields index, so we need to split the keys
+           ;; and represent it as a list of field names
+           (map #(if (str/includes? % ".")
+                   {:type  :nested-column-index
+                    :value (str/split % #"\.")}
+                   {:type  :normal-column-index
+                    :value %}))
+           set))))
+
+(defn- sample-documents [^MongoDatabase db table sort-direction]
+  (let [coll (mongo.util/collection db (:name table))]
+    (mongo.util/do-find coll {:limit metadata-queries/nested-field-sample-limit
+                             :skip 0
+                             :sort-criteria {:_id sort-direction}
```

Comment:
I've used map instead of ordered map here because there is only one key set.

I've updated the `do-find` function handle either ordered map or sequence of kv pairs as argument for `:sort-criteria`. Also the docstring was upadted accordingly.